crates/corro-types/src/sync.rs

```diff
@@ -85,8 +85,72 @@ pub struct SyncStateV1 {
     #[speedy(default_on_eof)]
     pub last_cleared_ts: Option<Timestamp>,
 }
-
 impl SyncStateV1 {
+    pub fn contains(&self, actor_id: &ActorId, versions: &RangeInclusive<Version>) -> bool {
+        !(self.heads.get(actor_id).cloned().unwrap_or_default() < *versions.end()
+            || self
+                .need
+                .get(actor_id)
+                .cloned()
+                .unwrap_or_default()
+                .iter()
+                .any(|x| x.start() < versions.end() && versions.start() < x.end()))
+    }
+
+    pub fn merge_needs(&mut self, needs: &HashMap<ActorId, Vec<SyncNeedV1>>) {
+        for (actor, need) in needs {
+            for n in need {
+                match n {
+                    SyncNeedV1::Full { versions } => {
+                        self.merge_full_version(*actor, versions);
+                    }
+                    SyncNeedV1::Partial { version, seqs } => {
+                        let mut delete = false;
+                        self.partial_need.entry(*actor).and_modify(|e| {
+                            if let Some(need_seqs) = e.get_mut(version) {
+                                let mut seqs_set =
+                                    RangeInclusiveSet::from_iter(need_seqs.clone().into_iter());
```

Comment:
Why not use `need_seqs` directly here? You wouldn't have to clone these twice if you mutate the entry directly.